### PR TITLE
Attr table - add/edit features in no geometry layers

### LINKED ISSUE #794: Add option to create and edit no geometry features
- Add toolbar at the bottom of the page with add option
  - add option should open preview panel on full screen
- Make preview panel fullscreen when clicking on a feature that does not have geometry

### PR DESCRIPTION
**Pull request by tomasMizera**
Added possibility to `add` and `edit` features without geometry.

When user selects layer in `browse data` (attribute table), there are three possible screens:

<table>
  <tr>
    <th>Features with geometry</th>
    <th>Features without geometry</th>
    <th>Features without geometry <br> (over limit *)</th>
  <tr>
    <td></td>
    <td></td>
    <td></td>
  </tr>
 </table>

***limit** is set to 10.000 ~ Input queries first 10.000 features from a layer and informs user when layer has more than this limit.

 + I used the same icon for no geometry features as in Qgis.
 + Clicking on `Add Feature` button opens classic feature form and saving/closing the form moves back to this view
 + Clicking on feature (without geo) opens fullscreen feature form  and saving/closing again leads back to this view
 + Toolbar to add feature is visible only for layers without geometry

Closes #794

### CODE REVIEW

File: app/qml/BrowseDataToolbar.qml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import QtQuick 2.0
```
Comment (PeterPetrik) [inline app/qml/BrowseDataToolbar.qml:1]: missing licence header?

File: app/qml/main.qml [hunk 2]
```diff
@@ -86,17 +89,23 @@ ApplicationWindow {
...
-        if (digitizing.isPairValid(pair)) {
-            digitizingHighlight.featureLayerPair = pair
-            digitizingHighlight.visible = true
-            featurePanel.show_panel(pair, "Add", "form")
-        } else {
-            popup.text = qsTr("Recording feature is not valid")
-            popup.open()
+    function saveRecordedFeature( pair, hasGeometry = true ) {
+
+      if ( !digitizing.isPairValid( pair ) && hasGeometry ) { // TODO: add check for no geometry pair validity
```
Comment (PeterPetrik) [inline app/qml/main.qml:94]: is this TODO still valid?

File: app/qml/main.qml [hunk 3]
```diff
@@ -86,17 +89,23 @@ ApplicationWindow {
...
+        popup.text = qsTr( "Recording feature is not valid" )
```
Comment (PeterPetrik) [inline app/qml/main.qml:95]: Recorded feature is ...

File: app/featuresmodel.cpp [hunk 4]
```diff
@@ -120,6 +121,8 @@ void FeaturesModel::emptyData()
 
   mFeatures.clear();
 
+  emit featuresCountChanged( mFeatures.size() );
```
Comment (PeterPetrik) [inline app/featuresmodel.cpp:124]: `emit featuresCountChanged(0);`

File: app/featuresmodel.h [hunk 5]
```diff
@@ -69,6 +74,7 @@ class FeaturesModel : public QAbstractListModel
 
     QList<QgsQuickFeatureLayerPair> mFeatures;
     Loader &mLoader;
+    int m_featuresCount;
```
Comment (PeterPetrik) [inline app/featuresmodel.h:77]: mFeaturesCount

File: app/featuresmodel.cpp [hunk 6]
```diff
@@ -150,3 +153,8 @@ Qt::ItemFlags FeaturesModel::flags( const QModelIndex &index ) const
 
   return Qt::ItemIsEditable;
 }
+
+int FeaturesModel::featuresCount() const
+{
+  return mFeatures.size();
```
Comment (PeterPetrik) [inline app/featuresmodel.cpp:159]: I am not sure if we do not want to display total number of features next to layer name? (@wonder-sk), not sure if displaying pagination limit is what user expects?

**Comment by PeterPetrik** [comment]
unrelated test failure (https://github.com/lutraconsulting/input/issues/813)

File: app/featuresmodel.cpp (hunk 6 above)
Comment (PeterPetrik) [reply app/featuresmodel.cpp:159]: @tomasMizera confirmed with Martin, should be total count of features in the layer

Comment (tomasMizera) [reply app/featuresmodel.cpp]: done

**Pull request merged**